### ISSUE TITLE
react/destructuring-assignment false positive(?) in assignment with ternary

### ISSUE THREAD

**Issue by William-Bryan** [yannickcr/eslint-plugin-react#2999]
```
function CollapsibleTableToggle(props) {
    const toggleClass = props.expanded ? 'fa fa-caret-down' : 'fa fa-caret-right';
    return <i className={`collapsible-table-toggle ${toggleClass}`} />;
}
```

The second line here produces a "Must use destructuring props assignment" error. Given that no object's properties are assigned to a variable, I would expect this to be fine. I can work around it by destructuring props in the function header, but this doesn't seem like the purpose of this error at all. Another function that uses `if (props.dangerouslySetInnerHTML)...` has no errors.

**Comment by ljharb** [comment]
The rule is saying that you must use assignment, because you’re not. In other words, no “props.expanded” - destructure props instead in the function signature.

**Comment by ljharb** [comment]
There may be a bug in the other function, though :-)

**Comment by ljharb** [comment]
Closing for now, since the code provided is warning as expected.
